.github/workflows/ruby.yml

```diff
@@ -305,3 +304,96 @@ jobs:
           $Env:MAKE = "ridk exec make"
           $ErrorActionPreference = "Continue"
           rake -f ruby\Rakefile
+
+  windows-msvc:
+    name: AMD64 Windows MSVC GLib
+    runs-on: windows-2019
+    if: ${{ !contains(github.event.pull_request.title, 'WIP') }}
+    timeout-minutes: 90
+    strategy:
+      fail-fast: false
+    env:
+      ARROW_BOOST_USE_SHARED: OFF
+      ARROW_BUILD_BENCHMARKS: OFF
+      ARROW_BUILD_SHARED: ON
+      ARROW_BUILD_STATIC: OFF
+      ARROW_BUILD_TESTS: OFF
+      ARROW_ACERO: ON
+      ARROW_DATASET: OFF
+      ARROW_FLIGHT: OFF
+      ARROW_FLIGHT_SQL: OFF
+      ARROW_HDFS: OFF
+      ARROW_HOME: "${{ github.workspace }}/dist"
```

Comment:
I had to change `ARROW_HOME` to be in the workspace directory rather than `/usr`, as `/usr` becomes `C:\Program Files\Git\usr` as a Windows path and I was getting errors when linking like:
```
[47/54] Linking target arrow-glib/arrow-glib-1700.dll
FAILED: arrow-glib/arrow-glib-1700.dll arrow-glib/arrow-glib-1700.pdb 
"link" @arrow-glib/arrow-glib-1700.dll.rsp
LINK : fatal error LNK1181: cannot open input file 'Files\Git\usr\bin.obj'
```

I didn't get to the bottom of where this was going wrong but it seems like something in meson is failing to quote the linker command inputs.